### PR TITLE
Adding timestamp to the template options' name

### PR DESCRIPTION
**Pull request by ygy**
- timeStamp replaces the previously used randId. 
- timeStamp uses the standard unix timestamp represented
  as a 8-char hex string.
- It represents the moment in time when the name is constructed. 
- It gives the possibility to search easily for instances, security
  groups, keypairs, etc based on timestamp without complicated
  enumeration

### CODE REVIEW

File: core/src/main/java/org/apache/brooklyn/core/location/cloud/names/BasicCloudMachineNamer.java [hunk 1]
```diff
@@ -41,9 +41,15 @@ protected String generateNewIdOfLength(ConfigBag setup, int len) {
...
-        shortener.append("randId", Identifiers.makeRandomId(4));
+        /* timeStamp replaces the previously used randId. 
+         * 
+         * timeStamp uses the standard unix timestamp represented as a 8-char hex string.
+         * 
+         * It represents the moment in time when the name is constructed. 
+         * It gives the possibility to search easily for instances, security groups, keypairs, etc
+         * based on timestamp without complicated enumeration        
+         */ 
+        shortener.append("timeStamp", Long.toString(System.currentTimeMillis() / 1000L, 16));
```
Comment (bostko) [inline core/src/main/java/org/apache/brooklyn/core/location/cloud/names/BasicCloudMachineNamer.java]: You could use Long.toString(1440680387, Character.MAX_RADIX) or even more explicitly Long.toString(l, 36)
This narrows down the required string length for the timestamp to 6

**Comment by CMoH** [comment]
I might be asking the wrong question here, but what will happen if more machines are created within the same second? Will they get the same name? Would this be a problem or not?

**Comment by ygy** [comment]
It will be highly unlikely to get 2 machines with one and the same name. 
The name is comprised by several other parameters like `appId` and `entityId`. They are random strings as well.
Most importantly jclouds itself append a randomly generated 8-char suffix at the end.

**Comment by CMoH** [comment]
I see - thanks for clarifying this.

File: core/src/main/java/org/apache/brooklyn/core/location/cloud/names/BasicCloudMachineNamer.java (hunk 1 above)
Comment (ygy) [inline core/src/main/java/org/apache/brooklyn/core/location/cloud/names/BasicCloudMachineNamer.java]: Thanks @bostko 
Addressed in commit 7fff655b05df098d9f35c97863da0d9902bf420a

**Comment by aledsage** [comment]
@ygy the build is failing at `CloudMachineNamerTest.testLengthMaxPermittedForMachineName` (see https://builds.apache.org/job/incubator-brooklyn-pull-requests/org.apache.brooklyn$brooklyn-core/1746/testReport/junit/org.apache.brooklyn.core.location.cloud/CloudMachineNamerTest/testLengthMaxPermittedForMachineName/). Can you take a look at fixing that please?

    java.lang.AssertionError: expected [10] but found [9]
    	at org.testng.Assert.fail(Assert.java:94)
    	at org.testng.Assert.failNotEquals(Assert.java:494)
    	at org.testng.Assert.assertEquals(Assert.java:123)
    	at org.testng.Assert.assertEquals(Assert.java:370)
    	at org.testng.Assert.assertEquals(Assert.java:380)
    	at org.apache.brooklyn.core.location.cloud.CloudMachineNamerTest.testLengthMaxPermittedForMachineName(CloudMachineNamerTest.java:118)

**Comment by aledsage** [comment]
Thanks @ygy - merging.

**Pull request merged**